os-server/src/main/java/app/openschool/course/CourseController.java

```diff
@@ -85,4 +95,92 @@ public ResponseEntity<Page<CourseDto>> findAll(
+              description = "Request object for creating new course")
+          @Valid
+          @RequestBody
+          CreateCourseRequest request) {
+    return ResponseEntity.status(HttpStatus.CREATED)
+        .body(CourseMapper.toCourseDto(courseService.add(request)));
+  }
+
+  @Operation(summary = "update course", security = @SecurityRequirement(name = "bearerAuth"))
+  @ApiResponses(
+      value = {
+        @ApiResponse(responseCode = "200", description = "Modifies the course and returns that"),
+        @ApiResponse(
+            responseCode = "400",
+            description = "Invalid request arguments supplied or not provided",
+            content = @Content(schema = @Schema(implementation = ResponseMessage.class))),
+        @ApiResponse(
+            responseCode = "403",
+            description = "Only users with ADMIN or MENTOR role have access to this method",
+            content = @Content(schema = @Schema()))
+      })
+  @PreAuthorize("hasAnyAuthority('ADMIN', 'MENTOR')")
+  @PutMapping("/{id}")
+  public ResponseEntity<CourseDto> update(
+      @Parameter(description = "Course id which will be updated") @PathVariable(value = "id")
+          Long courseId,
+      @io.swagger.v3.oas.annotations.parameters.RequestBody(
+              description = "Request object for updating the course")
+          @Valid
+          @RequestBody
+          UpdateCourseRequest request,
+      Principal principal) {
+    Course course = courseService.findCourseById(courseId).orElseThrow(IllegalAccessError::new);
+    if (!principal.getName().equals(course.getMentor().getEmail())) {
+      throw new IllegalArgumentException();
+    }
+    return ResponseEntity.ok()
+        .body(CourseMapper.toCourseDto(courseService.update(courseId, request)));
+  }
+
+  @Operation(summary = "delete course", security = @SecurityRequirement(name = "bearerAuth"))
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "204",
+            description = "The course was deleted",
+            content = @Content(schema = @Schema())),
+        @ApiResponse(
+            responseCode = "400",
+            description = "Invalid course id supplied",
+            content = @Content(schema = @Schema(implementation = ResponseMessage.class))),
+        @ApiResponse(
+            responseCode = "403",
+            description = "Only users with ADMIN or MENTOR role have access to this method",
+            content = @Content(schema = @Schema()))
+      })
+  @PreAuthorize("hasAnyAuthority('ADMIN', 'MENTOR')")
+  @DeleteMapping("/{id}")
+  public ResponseEntity<HttpStatus> delete(
+      @Parameter(description = "Id of course which will be deleted") @PathVariable(value = "id")
+          Long courseId,
+      Principal principal) {
+    Course course =
+        courseService.findCourseById(courseId).orElseThrow(IllegalArgumentException::new);
+    if (!course.getMentor().getEmail().equals(principal.getName())) {
```

Comment:
could you please make this a separate method and call everywhere, this logic is repeating in most of the endpoints